path/tree.go

```diff
@@ -93,49 +113,64 @@ func (t *tree) Find(path []string, index int) interface{} {
 	if index >= len(path) {
 		return t.value
 	}
-	node, _ := t.findNode(path, index, 0)
+	node, _ := t.findNode(path, index, 0, nil)
 	if node != nil {
 		return node.value
 	}
 	return nil
 }
 
-func (t *tree) findNode(path []string, index, matchCount int) (found *tree, foundMatchCount int) {
+func (t *tree) FindAndExtract(path []string, index int) (val interface{}, varMap map[string]interface{}) {
```

Comment:
Comment for the method.

Reply:
It's right there on the interface, but ok, I'll repeat it. :)